docs/key-concepts.md

```diff
@@ -0,0 +1,122 @@
+| 2024-02-11T09:12:45Z | Disk usage exceeded threshold           | WARNING  | Server: db-03, Disk Usage: 85%, Threshold: 80%     |
+| 2024-02-11T10:20:30Z | Critical error in database connection   | ERROR    | Server: db-02, Error Code: 500, Connection Timeout |
+| 2024-02-11T11:05:15Z | User authentication failed              | ERROR    | Username: johndoe, IP: 192.168.1.100               |
+| 2024-02-11T12:30:00Z | Scheduled job completed successfully    | INFO     | Job: Backup, Duration: 00:45:00                    |
+
+### Log ingestion
+
+As the name suggests, it refers to the process of **collecting and importing log data** from various sources into a centralized system for analysis and storage. It involves capturing log events generated by applications, servers, and other components, parsing and enriching the log data, and forwarding it to a centralized logging system or storage backend. Log ingestion systems often support various protocols and formats for data collection, including syslog, JSON, and structured logging.
+
+## 2. Metrics 📊
+
+Metrics are quantitative measurements that provide insight into the **performance** and **health** of a system. Unlike logs, which provide detailed event-based information, metrics focus on numerical data points over time. Common metrics include **CPU usage**, **memory utilization**, **response times**, **throughput**, and **error rates**.
+
+Metrics play a vital role in overseeing system performance, pinpointing irregularities, and uncovering patterns that could signify either emerging problems or chances for optimization. These metrics are usually gathered and consolidated from diverse origins within the system, including applications, servers, and infrastructure components.
+
+## 3. Traces 🔍
+
+Traces are a record of the execution path of a request as it moves through a distributed system. They provide detailed information about each step taken by the request, including timestamps, service interactions, and any errors encountered along the way. 
+
+Traces help developers and operators understand how requests flow through their systems, making it easier to identify bottlenecks, diagnose problems, and optimize performance.
+
+### Instrumentation for traces
+
+Instrumentation for traces involves modifying your code to add specific instructions that capture information about how requests move through your system-specific libraries or frameworks to generate trace data as requests flow through a distributed system. This instrumentation captures details such as request origins, service interactions, and timing information at various points within the system. It typically requires adding **code snippets** or **middleware** to applications to generate and propagate trace information across different services. Examples of instrumentation libraries commonly used for tracing include OpenTelemetry, Jaeger, Zipkin, and Datadog APM.
```

Comment:
`This instrumentation captures details such as request origins, service interactions, and timing information at various points within the system`
Let's keep these as bullet points.

`Examples of instrumentation libraries commonly used for tracing include OpenTelemetry, Jaeger, Zipkin, and Datadog APM.` 
I think we shouldn't mention about the other tools, since SigLens is an alternative for them. Let's just talk about OpenTelemetry

Reply:
Okay, I'll convert them to bullet points.

Do you think we should mention OTEL as well as Siglens?